File: src/main/java/by/architecture_map/belarus/service/ArticleService.java
package by.architecture_map.belarus.service;

import by.architecture_map.belarus.entity.Article;

import java.util.List;

public interface ArticleService {

    Article create(Article article);

    Article find(int id);

    /**
     * Using for finding Articles in Elasticsearch
     */
    List<Article> find(String request);

    List<Article> findAll();

    Article update(int id, Article updatedArticle);

    Article patchUpdate(int id, Article updatedArticle);

    void delete(int id);

}

File: src/main/kotlin/by/architecture_map/belarus/service/ArticleService.kt
package by.architecture_map.belarus.service

import by.architecture_map.belarus.entity.Article

interface ArticleService {

    fun create(article: Article): Article
    fun find(id: Int): Article

    /**
     * Using for finding Articles in Elasticsearch
     */
    fun find(request: String): List<Article>
    fun findAll(): List<Article>
    fun update(id: Int, updatedArticle: Article): Article
    fun patchUpdate(id: Int, updatedArticle: Article): Article
    fun delete(id: Int)
}
